code/game/machinery/computer/medical.dm

```diff
@@ -1,509 +1,258 @@
-
-			if (href_list["del_r2"])
-				QDEL_NULL(active2)
-
-			if (href_list["d_rec"])
-				var/datum/data/record/R = locate(href_list["d_rec"])
-				var/datum/data/record/M = locate(href_list["d_rec"])
-				if (!( GLOB.data_core.general.Find(R) ))
-					src.temp = "Record Not Found!"
+				if(!user_id_card)
 					return
-				for(var/datum/data/record/E in GLOB.data_core.medical)
-					if ((E.fields["ref"] == R.fields["ref"] || E.fields["id"] == R.fields["id"]))
-						M = E
-				src.active1 = R
-				src.active2 = M
-				src.screen = 4
-
-			if (href_list["new"])
-				if ((istype(src.active1, /datum/data/record) && !( istype(src.active2, /datum/data/record) )))
-					var/datum/data/record/R = new /datum/data/record(  )
-					R.fields["name"] = src.active1.fields["name"]
-					R.fields["id"] = src.active1.fields["id"]
-					R.name = text("Medical Record #[]", R.fields["id"])
-					R.fields["b_type"] = "Unknown"
-					R.fields["mi_dis"] = "None"
-					R.fields["mi_dis_d"] = "No minor disabilities have been declared."
-					R.fields["ma_dis"] = "None"
-					R.fields["ma_dis_d"] = "No major disabilities have been diagnosed."
-					R.fields["alg"] = "None"
-					R.fields["alg_d"] = "No allergies have been detected in this patient."
-					R.fields["cdi"] = "None"
-					R.fields["cdi_d"] = "No diseases have been diagnosed at the moment."
-					R.fields["notes"] = "No notes."
-					GLOB.data_core.medical += R
-					src.active2 = R
-					src.screen = 4
-
-			if (href_list["add_c"])
-				if (!( istype(src.active2, /datum/data/record) ))
-					return
-				var/a2 = src.active2
-				var/t1 = copytext(trim(strip_html(input("Add Comment:", "Med. records", null, null)  as message)),1,MAX_MESSAGE_LEN)
-				if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.is_mob_restrained() || (!in_range(src, usr) && (!isRemoteControlling(usr))) || src.active2 != a2))
+				if(ishuman(user))
+					user_id_card.forceMove(user.loc)
+					if(!user.get_active_hand())
+						user.put_in_hands(user_id_card)
+				else
+					user_id_card.forceMove(loc)
+				user_id_card = null
+		if("logout")
+			visible_message("[SPAN_BOLD("[src]")] states, \"AUTH LOGOUT: Session end confirmed.\"")
+			authenticated = FALSE
+			if(ishuman(user))
+				user_id_card.forceMove(user.loc)
+				if(!user.get_active_hand())
+					user.put_in_hands(user_id_card)
+			else
+				user_id_card.forceMove(loc)
+			user_id_card = null
+			return TRUE
+		if("print")
+			if(!authenticated || !target_id_card)
+				return
+
+			if(!COOLDOWN_FINISHED(src, print_cooldown))
+				visible_message("[SPAN_BOLD("[src]")] states, \"PRINT ERROR: system is still on cooldown.\"")
+				return
+			COOLDOWN_START(src, print_cooldown, PRINT_COOLDOWN_TIME )
+			print_paper()
+
+		if("eject")
+			if(target_id_card)
+				if(ishuman(user))
+					target_id_card.forceMove(user.loc)
+					if(!user.get_active_hand())
+						user.put_in_hands(target_id_card)
```

Comment:
put_in_hands already drops stuff on the floor if both hands are busy and you dont need to check it.

Reply:
(and drop it yourself)

Reply:
I think it's fine as is.